flask_dance/contrib/twitch.py

```diff
@@ -0,0 +1,91 @@
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+from functools import partial
+from flask.globals import LocalProxy, _lookup_app_object
+import os
+
+from flask import _app_ctx_stack as stack
+
+
+__maintainer__ = "Kerry Hatcher <kerry.hatcher@protonmail.com>"
+
+
+class TwitchBlueprint(OAuth2ConsumerBlueprint):
+    """
+    Add client-id to the API request headers
+    """
+
+    def session_created(self, session):
+        session.headers.update({"client-id": os.getenv("TWITCH_CLIENTID")})
```

Comment:
Rather than making this pull from an environment variable, it should pull from `app.config`. You can use the [`current_app`](https://flask.palletsprojects.com/en/1.1.x/api/#flask.current_app) proxy to make that easier.

I'm also not sure if it makes sense to do this in the `session_created()` callback, or if it makes sense to override `request()` and do it there. The former happens only once, and if the configured value changes, it won't pick it up; the latter will always pull it for every request, so it will pick up updates.

Regardless, this _definitely_ needs to be covered in the automated tests. Your tests need to set this value in the app config, make an API request, and assert that the header is properly set. You should also think about what should happen if this value is not provided. If Twitch won't allow you to use their API without this header, then I suggest throwing an exception if this value is not provided. You should have an automated test for that, too.

Reply:
YES! That makes a ton of sense!! Thanks for the guidance, I jumped straight in the deep end on this. Give me a day or so and I'll get all the changes in. 

As far as overriding `request()` goes I don't see an obvious use case for the client-id to change in a running app but that would be the most flexible for sure. Do you happen to have an example of that in another provider?

Reply:
I think I got it... resolving this for now.